satpy/tests/test_readers.py

```diff
@@ -662,6 +671,28 @@ def test_available_readers(self):
             self.assertIn('name', reader_info)
         self.assertEqual(reader_infos, sorted(reader_infos, key=lambda reader_info: reader_info['name']))
 
+    def test_available_readers_base_loader(self):
+        """Test the 'available_readers' function for yaml loader type BaseLoader."""
+        import yaml
+
+        from satpy import available_readers
+        from satpy._config import glob_config
+
+        def patched_import_error(name, globals=None, locals=None, fromlist=(), level=0):
+            if name in ('netcdf4', ):
+                raise ImportError(f"Mocked import error {name}")
+            return real_import(name, globals=globals, locals=locals, fromlist=fromlist, level=level)
+
+        self.monkeypatch.delitem(sys.modules, 'netcdf4', raising=False)
+        self.monkeypatch.setattr(builtins, '__import__', patched_import_error)
+
+        with pytest.raises(ImportError):
+            import netcdf4  # noqa: F401
+
+        reader_names = available_readers(yaml_loader=yaml.BaseLoader)
+        self.assertIn('abi_l1b', reader_names)  # needs netcdf4
```

Comment:
Well to be honest I got this monkey patching idea from the web. From my understanding the monkey patch should patch all imports of netcdf4 even in xarray and throw an ImportError.

Reply:
So...I'm not questioning the monkey patching, I'm saying that `abi_l1b` is one reader that won't be affected by this failed import, at least not at import time.

Reply:
ah ok, missunderstanding then. I added the `viirs_l1b`. Missed that one because I copied the line from above but didn't change it. Will push the change in a minute.